pkg/cmd/server/origin/master_config.go

```diff
@@ -415,6 +389,182 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
+		// be used to change the trusted signers for every pod anyway, so we're already effectively trusting it.
+		if len(ret.ServiceAccountTokenControllerOptions.RootCA) > 0 {
+			ret.ServiceAccountTokenControllerOptions.ServiceServingCA = append(ret.ServiceAccountTokenControllerOptions.ServiceServingCA, ret.ServiceAccountTokenControllerOptions.RootCA...)
+			ret.ServiceAccountTokenControllerOptions.ServiceServingCA = append(ret.ServiceAccountTokenControllerOptions.ServiceServingCA, []byte("\n")...)
+		}
+		ret.ServiceAccountTokenControllerOptions.ServiceServingCA = append(ret.ServiceAccountTokenControllerOptions.ServiceServingCA, serviceServingCA...)
+	}
+
+	ret.ServiceAccountControllerOptions = origincontrollers.ServiceAccountControllerOptions{
+		ManagedNames: options.ServiceAccountConfig.ManagedNames,
+	}
+
+	storageVersion := options.EtcdStorageConfig.OpenShiftStorageVersion
+	groupVersion := schema.GroupVersion{Group: "", Version: storageVersion}
+	annotationCodec := kapi.Codecs.LegacyCodec(groupVersion)
+
+	imageTemplate := variable.NewDefaultImageTemplate()
+	imageTemplate.Format = options.ImageConfig.Format
+	imageTemplate.Latest = options.ImageConfig.Latest
+
+	ret.BuildControllerConfig = origincontrollers.BuildControllerConfig{
+		DockerImage:           imageTemplate.ExpandOrDie("docker-builder"),
+		STIImage:              imageTemplate.ExpandOrDie("sti-builder"),
+		AdmissionPluginConfig: options.AdmissionConfig.PluginConfig,
+		Codec: annotationCodec,
+	}
+
+	vars, err := getOpenShiftClientEnvVars(options)
+	if err != nil {
+		return nil, err
+	}
+	ret.DeployerControllerConfig = origincontrollers.DeployerControllerConfig{
+		ImageName:     imageTemplate.ExpandOrDie("deployer"),
+		Codec:         annotationCodec,
+		ClientEnvVars: vars,
+	}
+	ret.DeploymentConfigControllerConfig = origincontrollers.DeploymentConfigControllerConfig{
+		Codec: annotationCodec,
+	}
+	ret.DeploymentTriggerControllerConfig = origincontrollers.DeploymentTriggerControllerConfig{
+		Codec: annotationCodec,
+	}
+
+	ret.ImageTriggerControllerConfig = origincontrollers.ImageTriggerControllerConfig{
+		HasBuilderEnabled: options.DisabledFeatures.Has(configapi.FeatureBuilder),
+		// TODO: make these consts in configapi
+		HasDeploymentsEnabled:  options.DisabledFeatures.Has("triggers.image.openshift.io/deployments"),
+		HasDaemonSetsEnabled:   options.DisabledFeatures.Has("triggers.image.openshift.io/daemonsets"),
+		HasStatefulSetsEnabled: options.DisabledFeatures.Has("triggers.image.openshift.io/statefulsets"),
+		HasCronJobsEnabled:     options.DisabledFeatures.Has("triggers.image.openshift.io/cronjobs"),
+	}
+	ret.ImageImportControllerOptions = origincontrollers.ImageImportControllerOptions{
+		MaxScheduledImageImportsPerMinute:          options.ImagePolicyConfig.MaxScheduledImageImportsPerMinute,
+		ResyncPeriod:                               10 * time.Minute,
```

Comment:
> wonder if we want to make the resyncPeriods configurable or we can just hardcode them in controllers

Ultimately, they should be configurable.  This is a move of existing hardcodedness.